File: Taller de Haskell/Problema 6/problema6.py
# Le pedimos el nombre y el género
nombre = input("Ingrese su nombre: ")
genero = input("Ingrese su género (Masculino/Femenino/Otro): ").lower()

# Lógica de la asignación de grupo
if genero == "femenino":
    grupo = "Grupo A"
elif genero == "masculino":
    grupo = "Grupo B"
else:
    grupo = "Grupo C"

print(nombre, "pertenece al", grupo)

File: Taller de Haskell/Problema 6/problema6.hs
import Data.Char (toLower)

asignarGrupo :: String -> String -> String
asignarGrupo nombre genero
    | genero == "femenino" = nombre ++ " pertenece al Grupo A"
    | genero == "masculino" = nombre ++ " pertenece al Grupo B"
    | otherwise = nombre ++ " pertenece al Grupo C"

main :: IO ()
main = do
    putStrLn "Ingrese su nombre:"
    nombre <- getLine
    putStrLn "Ingrese su género (Masculino/Femenino/Otro):"
    genero <- getLine
    putStrLn (asignarGrupo nombre (map toLower genero))
